Fipe_Challenge/Features/Detail/View/DetailViewController.swift

```diff
@@ -0,0 +1,109 @@
+import UIKit
+
+class DetailViewController: UIViewController {
+
+    var details: CarDetail?
+    var brandId: String?
+    var modelId: String?
+    var yearId: String?
+    var viewModel = DetailViewModel()
+    
+    lazy var viewCentral: UIView = {
+        let view = UIView()
+        view.backgroundColor = .lightGray
+        view.layer.cornerRadius = 10
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    lazy var brandLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = .systemFont(ofSize: 24)
+        return label
+    }()
+    
+    lazy var modelLabel: UILabel = {
+        let modelLabel = UILabel()
+        modelLabel.translatesAutoresizingMaskIntoConstraints = false
+        modelLabel.font = .systemFont(ofSize: 24)
+        modelLabel.numberOfLines = 0
+       return modelLabel
+    }()
+    
+    lazy var yearLabel: UILabel = {
+        let yearLabel = UILabel()
+        yearLabel.translatesAutoresizingMaskIntoConstraints = false
+        yearLabel.font = .systemFont(ofSize: 24)
+        return yearLabel
+    }()
+    
+    lazy var priceLabel: UILabel = {
+        let priceLabel = UILabel()
+        priceLabel.translatesAutoresizingMaskIntoConstraints = false
+        priceLabel.font = .boldSystemFont(ofSize: 30)
+        return priceLabel
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        view.addSubview(viewCentral)
+        viewCentral.addSubview(brandLabel)
+        viewCentral.addSubview(modelLabel)
+        viewCentral.addSubview(yearLabel)
+        viewCentral.addSubview(priceLabel)
+
```

Comment:
vc usou um padrão em uma controller, e outro aqui, seria interessante vc manter a coerência do código:

utilize sempre esse padrão :
```

override func viewDidLoad() {
      setupViews
}


// para setar as informacoes de views e como elas serao configuradas
setupViews() {
    view.backgroundColor = .white
    setupViewHierarchy()
    setupConstraints()
    
    // aqui vc pode setar tb configuracoes de navigation
}

// para hierarquia das views (ou seja ordem como elas vão aparecer
private setupViewHierarchy() {
 view.addSubview(viewCentral)
        viewCentral.addSubview(brandLabel)
        viewCentral.addSubview(modelLabel)
        viewCentral.addSubview(yearLabel)
        viewCentral.addSubview(priceLabel)
}

// para as constraints da view
private setupConstraints() {
NSLayoutConstraint.activate([
            viewCentral.centerXAnchor.constraint(equalTo: view.centerXAnchor),
            viewCentral.centerYAnchor.constraint(equalTo: view.centerYAnchor),
            viewCentral.widthAnchor.constraint(equalToConstant: 390),
            viewCentral.heightAnchor.constraint(equalToConstant: 230),

            brandLabel.topAnchor.constraint(equalTo: viewCentral.topAnchor, constant: 25),
            brandLabel.leadingAnchor.constraint(equalTo: viewCentral.leadingAnchor, constant: 10),
            brandLabel.trailingAnchor.constraint(equalTo: viewCentral.trailingAnchor, constant: 10),

            modelLabel.topAnchor.constraint(equalTo: brandLabel.bottomAnchor, constant: 10),
            modelLabel.leadingAnchor.constraint(equalTo: viewCentral.leadingAnchor, constant: 10),
            modelLabel.trailingAnchor.constraint(equalTo: viewCentral.trailingAnchor, constant: 10),

            yearLabel.topAnchor.constraint(equalTo: modelLabel.bottomAnchor, constant: 10),
            yearLabel.leadingAnchor.constraint(equalTo: viewCentral.leadingAnchor, constant: 10),
            yearLabel.trailingAnchor.constraint(equalTo: viewCentral.trailingAnchor, constant: 10),

            priceLabel.topAnchor.constraint(equalTo: yearLabel.bottomAnchor, constant: 25),
            priceLabel.centerXAnchor.constraint(equalTo: viewCentral.centerXAnchor)
        ])
}
```

Reply:
Outra sugestão q coloquei aqui e é bom q vc dê uma lida sobre, importância de controle de acesso dos métodos e properties dentro de uma class/struct. Me refiro a vc setar public/private/internal
Um pouco de doc aqui -> https://docs.swift.org/swift-book/documentation/the-swift-programming-language/accesscontrol/
aqui -> https://www.alura.com.br/artigos/ios-swift-como-funcionam-modificadores-de-acesso

Em resumo aqui nessa class, n faz muito sentido pra mim permitir que outras classes que estejam porventura herdando essa controller ou mesmo construindo ela em outro lugar, possam alterar métodos dentro dela. Quando seto private para meus métodos e properties, estou informando que apenas a minha classe pode ver e modificar informações ali dentro desses métodos, e para as variáveis elas só poderão ser lidas (get), não modificadas (set), a não ser que isso seja um requerimento do q vc vá fazer, daí nós podemos utilizar métodos públicos ou delegates para passar informação pra frente.